File: new/dart-maor/ayin-advance-label.dart
/// חוט · ayin-advance-label — תווית הכפתור-החכם לפי שלב.
/// המרה נאמנה מ-new/atoms/ayin-advance-label.mjs (חוק-4: המקור קדוש).
/// שקע: stageLabel(cfg, st) ⇒ תווית-השלב; a = פרטי-ה"עין" (stage/answerPushed).
/// אפס import (רק dart-core). התנהגות זהה-לחלוטין ל-JS.
String ayinAdvanceLabel(
  dynamic cfg,
  Map<String, dynamic> a,
  String Function(dynamic cfg, String st) stageLabel,
) {
  final st = a['stage'];
  if (st == 'new') return stageLabel(cfg, 'lead') + ' ←';
  if (st == 'lead') return '✓ אישור — ' + stageLabel(cfg, 'lead');
  if (st == 'eyes') return stageLabel(cfg, 'answer') + ' ←';
  if (st == 'answer') {
    return a['answerPushed'] == true
        ? '✓ ' + stageLabel(cfg, 'done')
        : '📞 דחיפה ללוח';
  }
  return '';
}

File: new/atoms/ayin-advance-label.mjs
/** חוט · ayin-advance-label — תווית הכפתור-החכם לפי שלב.
 *  חוזה: ayin-advance-label.contract.md · שקע: stageLabel
 *  חולץ כלשונו מ-maor/src/lib/ayin.ts:154-174 (קריאת-השכן שוקעה). */
export function ayinAdvanceLabel(cfg, a, stageLabel, T) {
    const st = a.stage;
    if (st === T.k1)
        return stageLabel(cfg, T.k2) + ' ←';
    if (st === T.k2)
        return T.k3 + stageLabel(cfg, T.k2);
    if (st === T.k4)
        return stageLabel(cfg, T.k5) + ' ←';
    if (st === T.k5)
        return a.answerPushed ? '✓ ' + stageLabel(cfg, T.k6) : T.k7;
    return '';
}
